Sources/ArcGISToolkit/Components/AR/TableTopSceneView.swift

```diff
@@ -0,0 +1,265 @@
+            plane.node.simdPosition = planeAnchor.center
+        }
+    }
+}
+
+/// A helper class to visualize a plane found by ARKit.
+class Plane: SCNNode {
+    let node: SCNNode
+    
+    init?(anchor: ARPlaneAnchor) {
+        // Create a node to visualize the plane's bounding rectangle.
+        let extent = SCNPlane(width: CGFloat(anchor.extent.x), height: CGFloat(anchor.extent.z))
+        node = SCNNode(geometry: extent)
+        node.simdPosition = anchor.center
+        
+        // `SCNPlane` is vertically oriented in its local coordinate space, so
+        // rotate it to match the orientation of `ARPlaneAnchor`.
+        node.eulerAngles.x = -.pi / 2
+        
+        super.init()
+        
+        node.opacity = 0.25
+        guard let material = node.geometry?.firstMaterial else { return nil }
+        
+        material.diffuse.contents = UIColor.white
+        
+        // Add the plane node as child node so they appear in the scene.
+        addChildNode(node)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+private extension ARSwiftUIViewProxy {
+    /// Performs a hit test operation to get the transformation matrix representing the corresponding real-world point for `screenPoint`.
+    /// - Parameter screenPoint: The screen point to determine the real world transformation matrix from.
+    /// - Returns: A `TransformationMatrix` representing the real-world point corresponding to `screenPoint`.
+    func hitTest(using screenPoint: CGPoint) -> TransformationMatrix? {
+        // Use the `raycastQuery` method on ARSCNView to get the location of `screenPoint`.
+        guard let query = raycastQuery(
+            from: screenPoint,
+            allowing: .existingPlaneGeometry,
+            alignment: .any
+        ) else { return nil }
+        
+        let results = session?.raycast(query)
+        
+        // Get the worldTransform from the first result; if there's no worldTransform, return nil.
+        guard let worldTransform = results?.first?.worldTransform else { return nil }
+        
+        // Create our hit test matrix based on the worldTransform location.
+        // right now we ignore the orientation of the plane that was hit to find the point
+        // since we only use horizontal planes, when we will start using vertical planes
+        // we should stop suppressing the quaternion rotation to a null rotation (0,0,0,1)
+        let hitTestMatrix = TransformationMatrix.normalized(
+            quaternionX: 0,
+            quaternionY: 0,
+            quaternionZ: 0,
+            quaternionW: 1,
+            translationX: Double(worldTransform.columns.3.x),
+            translationY: Double(worldTransform.columns.3.y),
+            translationZ: Double(worldTransform.columns.3.z)
+        )
+        
+        return hitTestMatrix
+    }
+}
+
+private extension SceneViewProxy {
```

Comment:
if we will need this for world scale then I would move it into the same file as the updateCamera extension method. Or that can be done when world scale needs it. It might only need the setFieldOfView method

Reply:
I think only `setFieldOfView()` is needed for world scale, but the method can be moved later if necessary.